backend-server/src/api/controllers/availabilityController.ts

```diff
@@ -38,3 +82,123 @@ export async function getAvailability(
     startTime
   );
 }
+
+export async function getAllAvailabilities(
+  organization: string,
+  interviewerUID: string
+) {
+  return await dataAccess.getAllAvailabilities(organization, interviewerUID);
+}
+
+export async function getAllCalendarAvailabilities(
+  organization: string,
+  interviewerUID: string
+): Promise<CalendarAvailability[]> {
+  const availabilities = (await dataAccess.getAllAvailabilities(
+    organization,
+    interviewerUID
+  )) as Availability[];
+  return makeMultipleCalendarAvailabilities(availabilities, organization);
+}
+
+export async function makeMultipleAvailabilities(
+  calendarAvailabilities: CalendarAvailability[],
+  organization: string
+): Promise<Availability[]> {
+  const availabilities: Availability[] = [];
+  for (const calendarAvailability of calendarAvailabilities) {
+    availabilities.push(
+      ...(await makeAvailabilitiesFromCalendarAvailability(
+        calendarAvailability,
+        organization
+      ))
+    );
+  }
+
+  return availabilities;
+}
+
+export async function makeMultipleCalendarAvailabilities(
+  availabilities: Availability[],
+  organization: string
+): Promise<CalendarAvailability[]> {
+  const interviewDuration = (
+    (await dataAccess.getOrganizationFields(organization)) as any
+  ).interviewDuration as number;
+  const calendarAvailabilities: CalendarAvailability[] = [];
+  for (let i = 0; i < availabilities.length; i++) {
+    const currStartDate = new Date(availabilities[i].startTime);
+    let currEndDate = add(currStartDate, { minutes: interviewDuration });
+    for (let j = i + 1; j < availabilities.length; j++) {
+      const nextStartDate = new Date(availabilities[j].startTime);
+      if (!isSameMinute(currEndDate, nextStartDate)) {
+        break;
+      }
+      currEndDate = add(nextStartDate, { minutes: interviewDuration });
+      i = j;
+    }
+    const newCalendarAvailability: CalendarAvailability = {
+      interviewerUID: availabilities[i].interviewerUID,
+      start: formatISO(currStartDate),
+      end: formatISO(currEndDate),
+    };
+    calendarAvailabilities.push(newCalendarAvailability);
+  }
+  console.log(calendarAvailabilities);
+
+  return calendarAvailabilities;
+}
+
+async function makeAvailabilitiesFromCalendarAvailability(
+  calendarAvailability: CalendarAvailability,
+  organization: string
+): Promise<Availability[]> {
+  const startDate: Date = new Date(calendarAvailability.start);
+  const endDate: Date = new Date(calendarAvailability.end);
+  const interviewDuration = (
```

Comment:
Can we rename this to `availabilityBlockLength` or something similar? Since it's not actually the interview duration, this might confuse us down the road. Same goes for `numIntervalsIn24Hours`, maybe we can rename that to `numAvailabilityBlocksIn24Hours`?